### PR TITLE
Use email_address table to check user's email when login with email address

### LINKED ISSUE #18547: Unable to login using e-mail (LDAP)
### Gitea Version

1.16.0

### Git Version

_No response_

### Operating System

_No response_

### How are you running Gitea?

Using gitea binary downloaded from dl.gitea.io.

### Database

PostgreSQL

### Can you reproduce the bug on the Gitea demo site?

No

### Log Gist

_No response_

### Description

When trying to log in using my e-mail, I get an error "Identifiant ou mot de passe invalide." (username or password invalid).
When using username, it works as expected.

### Screenshots

_No response_

### PR DESCRIPTION
**Pull request by lunny**
It's not clear what's the problem of the bug, but I think we should get the data from email_address table but not from user table now. It's amazing the test is wrong from 4 years ago.

Fix #18547

### CODE REVIEW

File: services/auth/signin.go [hunk 1]
```diff
@@ -24,17 +24,18 @@ import (
 func UserSignIn(username, password string) (*user_model.User, *auth.Source, error) {
 	var user *user_model.User
 	if strings.Contains(username, "@") {
-		user = &user_model.User{Email: strings.ToLower(strings.TrimSpace(username))}
+		emailAddress := user_model.EmailAddress{Email: strings.ToLower(strings.TrimSpace(username))}
```
Comment (zeripath) [inline services/auth/signin.go]: I think we need to use `email_address.lower_email` here.

Comment (zeripath) [reply services/auth/signin.go]:
```suggestion
		emailAddress := user_model.EmailAddress{LowerEmail: strings.ToLower(strings.TrimSpace(username))}
```

Comment (zeripath) [reply services/auth/signin.go:27]: Also why not query the is_activated at the same time?

**Review by zeripath — APPROVED** [review]

**Comment by johanvdw** [comment]
I think this is a useful addition (I can now login with different email addresses than the main email address is using local authentication. However, it does not fix https://github.com/go-gitea/gitea/issues/18547

If I set login_source to ldap I'm still not able to login using email, nor using username if the username is not equal to the login_name (Authentication Sign-In Name).

No error is logged.

**Comment by lunny** [comment]
replaced by #18789. @zeripath Could you review that PR again?

**Pull request closed without merging**